### PR TITLE
Response attribute for chat from messages operation and refactoring for bigdecimal usage

### PR DESCRIPTION
**Pull request by ayush-salesforce**

### CODE REVIEW

File: src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java [hunk 1]
```diff
@@ -192,36 +201,40 @@ private static List<String> createCorpusList(String filePath, String fileType, S
...
-    double dotProduct = 0.0;
-    double normA = 0.0;
-    double normB = 0.0;
-    for (int i = 0; i < vec1.length(); i++) {
-      double a = vec1.getDouble(i);
-      double b = vec2.getDouble(i);
-      dotProduct += a * b;
-      normA += Math.pow(a, 2);
-      normB += Math.pow(b, 2);
+  private static BigDecimal calculateCosineSimilarity(List<BigDecimal> embeddingList, List<BigDecimal> corpusEmbedding) {
```
Comment (sf-vishwastyagi) [inline src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java:204]: I don't think, we need BigDecimal here. The generated embeddings are  0.000002239895, -0.0038747797, 0.008896385 etc. The double can accurately handle the magnitude and precision without losing information. There may be minimal floating-point errors but the cumulative error in a typical cosine similarity calculation is unlikely to affect. BigDecimal would significantly slow down computations compared to double. Cosine similarity involves multiple multiplications, additions, and a square root, all of which are computationally heavier with BigDecimal.

File: src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java [hunk 2]
```diff
@@ -107,78 +116,78 @@ public static String executeTools(String originalPrompt, String prompt, String f
...
+      response = executeEinsteinRequest(accessTokeDTO, finalPayload, paramDetails.getModelApiName(), URI_MODELS_API_GENERATIONS);
 
     }
     return response;
   }
 
-  private static String executeEinsteinRequest(String accessToken, String payload, String modelName, String resource) {
-    String urlString = URL_BASE + modelName + resource;
-    return executeREST(accessToken, payload, urlString);
+  private static String executeEinsteinRequest(OAuthResponseDTO accessTokenDTO, String payload, String modelName,
```
Comment (sf-vishwastyagi) [inline src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java:125]: lets keep all private methods below all public methods

File: src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java [hunk 3]
```diff
@@ -107,78 +116,78 @@ public static String executeTools(String originalPrompt, String prompt, String f
...
+                                               String resource) {
+    String urlString = accessTokenDTO.getApiInstanceUrl() + URI_MODELS_API + modelName + resource;
```
Comment (sf-vishwastyagi) [inline src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java:127]: We should log this url, this may help for debugging purpose

**Review by sf-vishwastyagi — CHANGES_REQUESTED** [review]
Added review

File: src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java (hunk 3 above)
Comment (ayush-salesforce) [reply src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java:127]: Done

File: src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java (hunk 1 above)
Comment (ayush-salesforce) [reply src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java:204]: Done. Changed to double

File: src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java (hunk 2 above)
Comment (ayush-salesforce) [reply src/main/java/com/mule/einstein/internal/helpers/PayloadHelper.java:125]: Done

**Review by sf-vishwastyagi — APPROVED** [review]
Looks good to me

**Pull request merged**